osu.Game.Rulesets.Taiko/Difficulty/Preprocessing/Colour/Data/ColourEncoding.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Difficulty.Preprocessing.Colour.Data
+{
+    /// <summary>
+    /// Encodes a list of <see cref="MonoEncoding"/>s.
+    /// <see cref="MonoEncoding"/>s with the same <see cref="MonoEncoding.RunLength"/> are grouped together.
+    /// </summary>
+    public class ColourEncoding
```

Comment:
I'd prefer naming it something else other than `ColourEncoding` to be honest (as it is quite confusing), maybe something like `MonoStreakEncoding` or `GroupedMonoEncoding`? 

Related to this, do you think we should just rename `payload`s to something similar to their class name (for example, `MonoStreakEncoding`, and `MonoStreakPayload`/`MonoStreak`)

Reply:
If you ask me, you could drop the `Encoding` suffixes entirely because they don't really bring anything to the table either. I think you could do something like:

* `MonoEncoding` => `MonoStreak`
* `ColourEncoding` => `AlternatingMonoPattern`
* `CoupledColourEncoding` => `RepeatingHitPatterns`

The names aren't still 100% conveying the details of what the particular patterns are, but (a) you can't get that without the names getting overly verbose and (b) I feel that the proposals on the right at least convey the structure a little better (you have `MonoStreak`s which are grouped into `AlternatingPattern`s, which then are recognised as even larger `RepeatingHitPatterns`).

Reply:
Can't really think of better naming for these so I just went with them, applied in 5dcd4ce7c52de316d3b2f8665ffccfc86c3b9ada and 51176e95772736807b05d0158f8a79d14018dada